src/3d/framegraph/qgsdepthentity.cpp

```diff
@@ -0,0 +1,45 @@
+/***************************************************************************
+  qgsdepthentity.cpp
+  --------------------------------------
+  Date                 : June 2024
+  Copyright            : (C) 2024 by Benoit De Mezzo
+  Email                : benoit dot de dot mezzo at oslandia dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsdepthentity.h"
+#include "moc_qgsdepthentity.cpp"
+
+#include <Qt3DRender/QParameter>
+#include <QUrl>
+
+#include "qgsframegraph.h"
+#include "qgsabstractrenderview.h"
+
+QgsDepthEntity::QgsDepthEntity( Qt3DRender::QTexture2D *texture, Qt3DRender::QLayer *layer, QNode *parent )
+  : QgsRenderPassQuad( layer, parent )
+{
+  setObjectName( "depthRenderQuad" );
+
+  QMatrix4x4 modelMatrix;
+  modelMatrix.setToIdentity();
+
+  // construct material
+
+  mTextureParameter = new Qt3DRender::QParameter( "depthTexture", texture );
+  mTextureTransformParameter = new Qt3DRender::QParameter( "modelMatrix", QVariant::fromValue( modelMatrix ) );
```

Comment:
I would suggest to remove this parameter completely given it is always identity matrix (and remove the use of `modelMatrix` from `depth_render.vert`)